### PR TITLE
Add ability to retrieve packages by type and retrieve package metadata

### PR DESCRIPTION
**Pull request by chrisdicarlo**
Add ability to retrieve list of packages by type.  Also adds a method to retrieve a specific package's metadata.

### CODE REVIEW

File: src/Packagist.php [hunk 1]
```diff
@@ -24,6 +24,20 @@ public function __construct(Client $client, $baseUrl = 'https://packagist.org')
...
+     *
+     * @return array
+     */
+    public function getPackagesByType($type)
+    {
+        if (empty($type)) {
+            throw new Exception('You must pass a non-empty value');
+        }
+
+        return $this->makeRequest('/packages/list.json', ['type' => $type]);
```
Comment (freekmurze) [inline src/Packagist.php]: use `compact($type)` here

**Review by freekmurze — CHANGES_REQUESTED** [review]

**Comment by freekmurze** [comment]
Could you also resolve any merge conflicted and update the readme?

File: src/Packagist.php (hunk 1 above)
Comment (chrisdicarlo) [reply src/Packagist.php]: Will do!

**Comment by chrisdicarlo** [comment]
Resolved all the conflicts, updated the README, and fixed that one typo.

**Review by freekmurze — APPROVED** [review]

**Comment by freekmurze** [comment]
Thanks!

**Pull request merged**